src/Data/Fin/Properties.agda

```diff
@@ -128,6 +131,10 @@ fromℕ-toℕ : ∀ {n} (i : Fin n) → fromℕ (toℕ i) ≡ strengthen i
 fromℕ-toℕ zero    = refl
 fromℕ-toℕ (suc i) = cong suc (fromℕ-toℕ i)
 
+≤fromℕ : ∀ {n} → (i : Fin (ℕ.suc n)) → i ≤ fromℕ n
+≤fromℕ {n} i = ℕₚ.≤-trans (ℕₚ.≤-pred (toℕ<n i))
+                           (ℕₚ.≤-reflexive (sym (toℕ-fromℕ n)))
```

Comment:
Using `subst` instead of `ℕₚ.≤-trans` should shorten the proof.

Reply:
Are you sure you'd get the same proof? `≤` has computational content and `subst` basically acts like a permutation in cases like this.

Reply:
Err I'm afraid I'm not quite sure what you mean? Yes it will change the computational behaviour of the proof (more efficient I think, though don't quote me on this) but the type of the resulting term will be the same no? What do you mean by "act like a permutation". A permutation on what?

Reply:
I don't know about performance, but 
```agda
subst (toℕ i ℕ.≤_) (sym (toℕ-fromℕ n)) (ℕₚ.≤-pred (toℕ<n i))
```
doesn't seem much better in terms of length?

Reply:
(I'm happy to change it anyway)

Reply:
What I meant above about `subst`: think of what `subst Fin _ (+-comm m n)` does: Not only does it change `Fin (n +m)` into `Fin (m + n)` [or the other way around?], but it does this by 'relabelling' the inhabitants of `Fin` (which is a no-op, I know). In a way, this is a lie. The witnesses that `Fin (n + m)` used to be 0..n-1 and n..n+m-1, but now 0..m-1 and m..m+n-1. In their totality, they are the same (and so the types are).  It's easiest to think with a concrete case, say n=2 and m=5. 

Perhaps a better explanation is to compare two different ways of doing this
1. `subst`
2. via the chain `Fin (n + m) ↔ Fin n ⊎ Fin m ↔ Fin m ⊎ Fin n ↔ Fin (m + n)` where the middle one actually **does** something (swaps `inj₁` and `inj₂`). That actually does permute things.

This is the sense in which `subst` is a permutation, though it is a 'virtual' permutation, that changes the meaning of the underlying elements without actually performing a computation. An elaborate, somewhat convenient, lie.